### DISCUSSION TITLE
Add gatsby cli for graphiql only

### DISCUSSION THREAD

**Discussion by webrgp** [gatsbyjs/gatsby#34105] [Ideas / Feature Requests]
## Summary

When I am in early stage development, connecting to 3rd party APIs, implementing queries, fragments, etc OR having to implement or debug existing queries or fragments, due to API changes, it would be nice to be able to run an command that would basically just run the GraphQL aspect of Gatsby.

### Suggested solution

The command could be something like `gatsby develop:graphql` or just `gatsby graphql`

```sh
$ gatsby develop:graphql
...
...
View GraphiQL, an in-browser IDE, to explore your site's data and schema
⠀
  http://localhost:8000/___graphql
⠀
Note that the development build is not optimized.
```

I think that would make easier to work on project that interact with big APIs without the clutter of related errors connected to other components, etc, or to implement type extension.

Right now I use a separate `gatsby-starter-minimal` just to work queries and implement on the project.

**Comment by ankurparihar** [comment]
Much needed

**Comment by ranjanashwin** [comment]
This is a great suggestion, and I can definitely relate to the use case — especially when working with large external APIs or debugging query structures early in a project.

Having a lightweight command like `gatsby graphql` or `gatsby develop:graphql` that spins up only the schema and GraphiQL interface without loading the full dev server would be very useful. It could streamline schema exploration and help reduce noise from unrelated plugin or component errors during early development stages.

A workaround I've used is running `gatsby develop` with minimal pages/components and relying heavily on GraphiQL at `localhost:8000/___graphql`, but a CLI flag or dedicated mode would be much cleaner.

Definitely worth considering for the CLI roadmap — would love to see this get traction.
